src/types/mapping.ts

```diff
@@ -0,0 +1,120 @@
+            const hentVerdi = () => {
+                if (spm.svartype == RSSvartype.BELOP) {
+                    return (Number(svar.verdi) / 100).toString()
+                }
+                return svar.verdi
+            }
+            return {
+                id: svar.id,
+                verdi: hentVerdi(),
+                avgittAv: svar.avgittAv,
+            }
+        }),
+    }
+    const undersporsmal = rsToSporsmal(spm.undersporsmal, spm.kriterieForVisningAvUndersporsmal, false)
+
+    const idtag = tag as keyof typeof TagTyper
+
+    return new Sporsmal(
+        spm.id,
+        TagTyper[idtag],
+        tagIndex,
+        spm.sporsmalstekst === null ? '' : spm.sporsmalstekst,
+        spm.undertekst,
+        spm.svartype as any as RSSvartype,
+        spm.min,
+        spm.max,
+        spm.pavirkerAndreSporsmal,
+        spm.kriterieForVisningAvUndersporsmal as any,
+        svarliste,
+        undersporsmal,
+        kriterie,
+        erHovedsporsmal,
+    )
+}
+
+export function rsToSporsmal(
+    spms: RSSporsmal[],
+    kriterie: RSVisningskriterieType | null,
+    erHovedsporsmal: boolean,
+): ReadonlyArray<Sporsmal> {
+    const sporsmals: Sporsmal[] = []
+    if (spms === undefined) {
+        return sporsmals
+    }
+    spms.forEach((rssp) => {
+        const spm: Sporsmal = skapSporsmal(rssp, kriterie, erHovedsporsmal)
+        sporsmals.push(spm)
+    })
+
+    if (
+        sporsmals.length >= 2 &&
+        sporsmals[sporsmals.length - 1].tag === TagTyper.VAER_KLAR_OVER_AT &&
+        sporsmals[sporsmals.length - 2].tag === TagTyper.BEKREFT_OPPLYSNINGER
+    ) {
+        // Det finnes tilfeller opprettet i db før 15 Mai 2020 hvor disse er i "feil rekkefølge" Dette fikser sorteringa
+        // Se også https://github.com/navikt/syfosoknad/commit/1983d32f3a7fb28bbf17126ea227d91589ad5f35
+        const tmp = sporsmals[sporsmals.length - 1]
+        sporsmals[sporsmals.length - 1] = sporsmals[sporsmals.length - 2]
+        sporsmals[sporsmals.length - 2] = tmp
+    }
+    return sporsmals
+}
+
+export function rsToSoknad(soknad: RSSoknad): Soknad {
+    const stat = soknad.status as keyof typeof RSSoknadstatus
+    const type = soknad.soknadstype as keyof typeof RSSoknadstype
+    let arbeidsgiver = undefined
+    if (soknad.arbeidsgiver) {
+        arbeidsgiver = {
+            navn: soknad.arbeidsgiver.navn,
+            orgnummer: soknad.arbeidsgiver.orgnummer,
+        }
+    }
+    return new Soknad(
+        soknad.id,
+        soknad.sykmeldingId || undefined,
+        RSSoknadstype[type],
+        RSSoknadstatus[stat],
+        soknad.arbeidssituasjon ? RSArbeidssituasjon[soknad.arbeidssituasjon] : undefined,
+        dayjsToDate(soknad.fom!)!,
+        dayjsToDate(soknad.tom!)!,
+        dayjsToDate(soknad.avbruttDato!)!,
+        dayjsToDate(soknad.opprettetDato!)!,
+        dayjsToDate(soknad.sendtTilNAVDato!)!,
+        dayjsToDate(soknad.sendtTilArbeidsgiverDato!)!,
```

Comment:
Funker det med ! her, siden disse kan være null. Burde den ikke ha sagt i fra 🤔

Reply:
Det er kanskje en TypeScript greie og ikke runtime

Reply:
jeg tror selve typinga her er litt tullete opprinnelig. Dette er bare flytta ut fra konstruktøren i egen mapping metode for at data klassen er renere